### PR TITLE
[Doc] Update Tutorial to use TypeScript by default

### PR DESCRIPTION
**Pull request by adguernier**
Update `js` and `jsx` snippets to `ts` and `tsx` in __Getting Started / Tutorial__

### CODE REVIEW

**Comment by fzaninotto** [comment]
I think you should change the code type to typescript in all snippets, even if you don't add annotations. Otherwise, users will wonder why some snippets have a JS/TS switcher and some don't.

File: docs/Tutorial.md [hunk 1]
```diff
@@ -877,13 +877,13 @@ For this tutorial, since there is no public authentication API, we can use a fak
...
+```ts
 // in src/authProvider.ts
+import { AuthProvider } from "react-admin";
 
-// TypeScript users must reference the type: `AuthProvider`
-export const authProvider = {
+export const authProvider: AuthProvider = {
   // called when the user attempts to log in
-  login: ({ username }) => {
+  login: ({ username }: { username: string }) => {
```
Comment (djhi) [inline docs/Tutorial.md:886]: As you already declared `authProvider` as an `AuthProvider`, you don't need to provide more types. Same for all methods

File: docs/Tutorial.md [hunk 2]
```diff
@@ -372,11 +372,11 @@ In react-admin, fields are just React components. When rendered, they grab the `
 
 That means that you can do the same to write a custom Field. For instance, here is a simplified version of the `<UrlField>`:
 
-```jsx
+```tsx
 // in src/MyUrlField.tsx
 import { useRecordContext } from "react-admin";
 
-const MyUrlField = ({ source }) => {
+const MyUrlField = ({ source }: { source: string }) => {
```
Comment (djhi) [inline docs/Tutorial.md:379]: You can import `FieldProps` from `react-admin` and use it here:
```suggestion
const MyUrlField = ({ source }: FieldProps) => {
```

File: docs/Tutorial.md [hunk 3]
```diff
@@ -415,13 +415,13 @@ The `<MyUrlField>` component is a perfect opportunity to illustrate how to custo
...
 {% raw %}
-```jsx
+```tsx
 // in src/MyUrlField.tsx
 import { useRecordContext } from "react-admin";
 import { Link } from "@mui/material";
 import LaunchIcon from "@mui/icons-material/Launch";
 
-const MyUrlField = ({ source }) => {
+const MyUrlField = ({ source }: { source: string }) => {
```
Comment (djhi) [inline docs/Tutorial.md:424]: Same

**Review by djhi — CHANGES_REQUESTED** [review]

File: docs/Tutorial.md (hunk 2 above)
Comment (fzaninotto) [reply docs/Tutorial.md:379]: You marked this as fixed but it isn't.

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/Tutorial.md (hunk 2 above)
Comment (adguernier) [reply docs/Tutorial.md:379]: As discussed with @djhi, typed params with `FieldProps` generates Typescript errors, so we decided to let it as is

Comment (djhi) [reply docs/Tutorial.md:379]: I'll work on a fix when I can

File: docs/Tutorial.md [hunk 4]
```diff
@@ -47,7 +47,7 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import { App } from './App';
 
-ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
+ReactDOM.createRoot(document.getElementById('root')!).render(
```
Comment (slax57) [inline docs/Tutorial.md:50]: Don't know how others feel about this change, but I liked to original code better personally

Comment (fzaninotto) [reply docs/Tutorial.md:50]: Yet we're based on Vite, and they did the same in https://github.com/vitejs/vite/commit/6fe14911deac34f5874687705a46deb9caf1cb10

**Review by fzaninotto — APPROVED** [review]

File: docs/Tutorial.md (hunk 4 above)
Comment (slax57) [reply docs/Tutorial.md:50]: alright, valid point :grin:

**Review by slax57 — APPROVED** [review]

**Pull request merged**